src/main/java/com/sourcery/gymapp/backend/workout/service/WorkoutStatsService.java

```diff
@@ -0,0 +1,141 @@
+        int differenceInWeight = totalWeightCurrentMonth - totalWeightPreviousMonth;
+
+        List<WorkoutStatsDto> userStats = new ArrayList<>();
+
+        if (totalWorkoutsCurrentMonth >= 1) {
+            String isPlural = totalWorkoutsCurrentMonth > 1 ? "s" : "";
+            userStats.add(new WorkoutStatsDto(
+                    UUID.randomUUID(),
+                    "totalWorkouts",
+                    "You have completed %d workout%s this month!"
+                            .formatted(totalWorkoutsCurrentMonth, isPlural)
+            ));
+        }
+
+        if (differenceInWorkouts >= 1) {
+            String isPlural = totalWorkoutsCurrentMonth > 1 ? "s" : "";
+            userStats.add(new WorkoutStatsDto(
+                    UUID.randomUUID(),
+                    "workoutDifference",
+                    "You have completed %d more workout%s than the last month!"
+                            .formatted(differenceInWorkouts, isPlural)
+            ));
+        }
+
+        if (totalWeightCurrentMonth > 0) {
+            userStats.add(new WorkoutStatsDto(
+                    UUID.randomUUID(),
+                    "totalWeight",
+                    "You have lifted a total of " + totalWeightCurrentMonth + " kg this month!"
+            ));
+
+            userStats.add(new WorkoutStatsDto(
+                    UUID.randomUUID(),
+                    "totalWeight",
+                    totalWeightCurrentMonth +" kg " + WeightComparisonUtil.getMessageByWeight(totalWeightCurrentMonth)
+            ));
+        }
+
+        if (differenceInWeight > 0) {
+            userStats.add(new WorkoutStatsDto(
+                    UUID.randomUUID(),
+                    "weightDifference",
+                    "You have lifted " + differenceInWeight + " kg more than the last month!"
+            ));
+        }
+        return userStats;
+    }
+
+    private int getWorkoutCount(Integer offsetMonth) {
+        UUID currentUserId = workoutCurrentUserService.getCurrentUserId();
+        List<ZonedDateTime> startAndEndOfTheMonth = offsetDateService.getMonthlyDateRangeOffset(offsetMonth);
+
+        return workoutRepository
+                .countWorkoutsByUserIdAndDateBetween(
+                        currentUserId,
+                        startAndEndOfTheMonth.getFirst(),
+                        startAndEndOfTheMonth.getLast()
+                );
+    }
+
+    private int getTotalWeight(Integer offsetMonth) {
+        UUID currentUserId = workoutCurrentUserService.getCurrentUserId();
+        List<ZonedDateTime> startAndEndOfTheMonth = offsetDateService.getMonthlyDateRangeOffset(offsetMonth);
+
+        Integer totalWeight = workoutRepository.getTotalWeightByUserIdAndDateBetween(
+                currentUserId,
+                startAndEndOfTheMonth.getFirst(),
+                startAndEndOfTheMonth.getLast()
+        );
+
+        return totalWeight != null ? totalWeight : 0;
```

Comment:
lets return Optional from the repository

(using Optional makes code safer, avoids NullPointerException, improves readability -> kinda when we see Optional we assume it might be null)